react/src/context/PostHogProvider.tsx

```diff
@@ -54,51 +52,35 @@ export function PostHogProvider({ children, client, apiKey, options }: WithOptio
         if (apiKey) {
-            // If the client was already initialized, we'll attempt to return the existing instance
-            // This is likely to occur when someone is developing locally under `React.StrictMode`
-            //
-            // There's no built-in way to detect React.StrictMode, so we set a flag to check if the client was already initialized
-            // and return the existing instance if it was.
-            //
-            // This might also be triggered if someone attempts to change their `apiKey` or
-            // set a different set of `options`. This is handled separately below.
-            if (alreadyInitialized !== false) {
-                // Changing the apiKey isn't well supported and we'll simply log a message suggesting them
-                // to take control of the `client` initialization themselves. This is tricky to handle
-                // ourselves because we wouldn't know if we should call `.reset()` or not, for example.
-                if (apiKey !== alreadyInitialized.previousAPIKey) {
-                    console.warn(
-                        "[PostHog.js] You have provided a different `apiKey` to `PostHogProvider` than the one that was already initialized. This is not supported by our provider and we'll keep using the previous key. If you need to toggle between API Keys you need to control the `client` yourself and pass it in as a prop rather than an `apiKey` prop."
-                    )
-                }
-
-                // Changing options is better supported because we can just call `posthogJs.set_config(options)`
-                // and they'll be good to go with their new config. The SDK will know how to handle the changes.
-                if (options && !isDeepEqual(options, alreadyInitialized.previousOptions)) {
-                    posthogJs.set_config(options)
-                }
+            // return the global client, we'll initialize it in the useEffect
+            return posthogJs
+        }
 
-                // Keep track of the possibly-new set of apiKey and options
-                setAlreadyInitialized({
-                    previousAPIKey: apiKey,
-                    previousOptions: options ?? {},
-                })
+        console.warn(
+            '[PostHog.js] No `apiKey` or `client` were provided to `PostHogProvider`. Using default global `window.posthog` instance. You must initialize it manually. This is not recommended behavior.'
+        )
+        return posthogJs
+    }, [client, apiKey, JSON.stringify(options)]) // Stringify options to be a stable reference
 
-                // Return the same already-initialized global client
-                return posthogJs
-            }
+    // TRICKY: The init needs to happen in a useEffect rather than useMemo, as useEffect does not happen during SSR. Otherwise
+    // we'd end up trying to call posthogJs.init() on the server, which can cause issues around hydration and double-init.
+    useEffect(() => {
```

Comment:
Since we have a `useEffect`, is there anything we should be doing when this provider is unmounted? Maybe we should call `reset`? Not sure if it makes sense as a default tho